src/rpc/rawtransaction.cpp

```diff
@@ -151,6 +152,67 @@ UniValue TxShieldedOutputsToJSON(const CTransaction& tx) {
     return vdesc;
 }
 
+UniValue TxActionsToJSON(const rust::Vec<orchard_bundle::Action>& actions)
+{
+    UniValue arr(UniValue::VARR);
+    for (const auto& action : actions) {
+        UniValue obj(UniValue::VOBJ);
+        auto cv = action.cv();
+        obj.pushKV("cv", HexStr(cv.begin(), cv.end()));
+        auto nullifier = action.nullifier();
+        obj.pushKV("nullifier", HexStr(nullifier.begin(), nullifier.end()));
+        auto rk = action.rk();
+        obj.pushKV("rk", HexStr(rk.begin(), rk.end()));
+        auto cmx = action.cmx();
+        obj.pushKV("cmx", HexStr(cmx.begin(), cmx.end()));
+        auto ephemeralKey = action.ephemeral_key();
+        obj.pushKV("ephemeralKey", HexStr(ephemeralKey.begin(), ephemeralKey.end()));
+        auto encCiphertext = action.enc_ciphertext();
+        obj.pushKV("encCiphertext", HexStr(encCiphertext.begin(), encCiphertext.end()));
+        auto outCiphertext = action.out_ciphertext();
+        obj.pushKV("outCiphertext", HexStr(outCiphertext.begin(), outCiphertext.end()));
+        auto spendAuthSig = action.spend_auth_sig();
+        obj.pushKV("spendAuthSig", HexStr(spendAuthSig.begin(), spendAuthSig.end()));
+        arr.push_back(obj);
+    }
+    return arr;
+}
+
+// See https://zips.z.cash/zip-0225
+// TODO proofs, bindingSig
+void TxOrchardBundleToJSON(const CTransaction& tx, UniValue& entry)
+{
+    auto bundle = tx.GetOrchardBundle().GetDetails();
+
+    // If this tx has no actions, then flags, value balance, etc. are not present.
+    auto num_actions = bundle->num_actions();
+    if (num_actions == 0) return;
+
+    UniValue obj(UniValue::VOBJ);
+    auto actions = bundle->actions();
+    obj.pushKV("actions", TxActionsToJSON(actions));
+    {
```

Comment:
Hmm, this means a v5 transaction JSON doesn't include an `orchard` block if there are no Actions. While it's true that these fields aren't present in the transaction itself in that case, we do have a note in the protocol spec that `valueBalanceOrchard` is defined to be zero if there are no Orchard Actions in a transaction (since it still needs to be used in whole-transaction calculations). And prior to this PR, that was the behaviour of `getrawtransaction` after #5969 was merged.

I think we should unconditionally have `actions`, `valueBalance`, and `valueBalanceZat` present in the `orchard` object for v5 transactions (so downstream parsers don't need to think about the no-action default, and that case is encoded explicitly as `"orchard": { actions: [], ... }` instead of implicitly as the lack of an `orchard` section, which matches how the API works for transparent, Sprout, and Sapling). Then if `actions` is not empty then we conditionally include `flags`, `anchor`, `proof`, and `bindingSig`. I'm okay with having `flags` come after `valueBalance*` in the JSON output to make the rendering easier.
```suggestion
    UniValue obj(UniValue::VOBJ);
    auto actions = bundle->actions();
    obj.pushKV("actions", TxActionsToJSON(actions));
    auto valueBalanceZat = bundle->value_balance_zat();
    obj.pushKV("valueBalance", ValueFromAmount(valueBalanceZat));
    obj.pushKV("valueBalanceZat", valueBalanceZat);

    // If this tx has no actions, then flags, anchor, etc. are not present.
    if (!actions.empty()) {
```